src/System.Text.Json/tests/JsonElementWriteTests.cs

```diff
@@ -1043,7 +1044,61 @@ public static void WriteValueInsideObject(bool skipValidation)
                     writer.WriteEndObject();
                     writer.Flush();
 
-                    AssertContents("{}", buffer);
+                    JsonTestHelper.AssertContents("{}", buffer);
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData(false, "\"message\"", "\"message\"", true)]
+        [InlineData(true, "\"message\"", "\"message\"", true)]
+        [InlineData(false, "\">><++>>>\\\">>\\\\>>&>>>\u6f22\u5B57>>>\"", "\">><++>>>\\\">>\\\\>>&>>>\u6f22\u5B57>>>\"", false)]
+        [InlineData(true, "\">><++>>>\\\">>\\\\>>&>>>\u6f22\u5B57>>>\"", "\">><++>>>\\\">>\\\\>>&>>>\u6f22\u5B57>>>\"", false)]
+        [InlineData(false, "\"mess\\r\\nage\\u0008\\u0001!\"", "\"mess\\r\\nage\\b\\u0001!\"", true)]
+        [InlineData(true, "\"mess\\r\\nage\\u0008\\u0001!\"", "\"mess\\r\\nage\\b\\u0001!\"", true)]
+        public static void WriteWithRelaxedEscaper(bool indented, string jsonIn, string jsonOut, bool matchesRelaxedEscaping)
+        {
+            var buffer = new ArrayBufferWriter<byte>(1024);
+            using (JsonDocument doc = JsonDocument.Parse($" [  {jsonIn}  ]", s_options))
+            {
+                JsonElement target = doc.RootElement[0];
+
+                {
+                    var options = new JsonWriterOptions
+                    {
+                        Indented = indented,
+                    };
+
+                    using var writer = new Utf8JsonWriter(buffer, options);
+
+                    target.WriteTo(writer);
+                    writer.Flush();
+
+                    if (matchesRelaxedEscaping)
+                    {
+                        JsonTestHelper.AssertContents(jsonOut, buffer);
+                    }
+                    else
+                    {
+                        JsonTestHelper.AssertContentsNotEqual(jsonOut, buffer);
+                    }
+                }
+
+                buffer.Clear();
+
+                {
+                    var options = new JsonWriterOptions
+                    {
+                        Indented = indented,
+                        Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+                    };
+
+                    using var writer = new Utf8JsonWriter(buffer, options);
```

Comment:
Personally, I vastly prefer "block statement using".  This using ends after AssertContents, but following the least-need principle the using should end after (or instead-of) writer.Flush. (The new using statement is particularly bad in async methods, where it can make the using straddle an await it didn't need to)

So, replacing all of the usings with scoped blocks would make me happier, but it's not critical.